app/models/chargeback.rb

```diff
@@ -80,30 +80,24 @@ def self.build_results_for_report_chargeback(options)
     vm_owners = vms.inject({}) { |h, v| h[v.id] = v.evm_owner_name; h }
     options[:ext_options] ||= {}
 
+    base_rollup = MetricRollup.includes(
+      :resource           => :hardware,
+      :parent_host        => :tags,
+      :parent_ems_cluster => :tags,
+      :parent_storage     => :tags,
+      :parent_ems         => :tags)
+                              .select(*Metric::BASE_COLS).order("resource_id, timestamp")
     perf_cols = MetricRollup.attribute_names
-    options[:ext_options][:only_cols] = Metric::BASE_COLS
-    rates = ChargebackRate.where(:default => true)
-    rates.each do |rate|
-      options[:ext_options][:only_cols] += rate.chargeback_rate_details.collect do |r|
-        r.metric if perf_cols.include?(r.metric.to_s)
-      end.compact
+    rate_cols = ChargebackRate.where(:default => true).flat_map do |rate|
+      rate.chargeback_rate_details.map(&:metric).select { |metric| perf_cols.include?(metric.to_s) }
     end
+    base_rollup = base_rollup.select(*rate_cols)
```

Comment:
They are cumulative.
I guess we could have built up `rate_cols` from both sections. I guess we could have also tacked on more columns into the select in the loop a few lines up.

Lets go with what ever is easier for you to read and less "clever".